numpy/_core/shape_base.py

```diff
@@ -455,6 +457,77 @@ def stack(arrays, axis=0, out=None, *, dtype=None, casting="same_kind"):
     return _nx.concatenate(expanded_arrays, axis=axis, out=out,
                            dtype=dtype, casting=casting)
 
+def _unstack_dispatcher(x, *, axis=None):
+    return (x,)
+
+@array_function_dispatch(_unstack_dispatcher)
+def unstack(x, /, *, axis=0):
+    """
+    Splits an array into a sequence of arrays along the given axis.
+
+    The ``axis`` parameter specifies the axis along which the array will be
+    split. of the new axis in the dimensions of the result. For example, if
+    ``axis=0`` it will be the first dimension and if ``axis=-1`` it will be
+    the last dimension.
+
+    The result is a tuple of arrays split along ``axis``. ``unstack`` serves
+    as the reverse operation of :py:func:`stack`, i.e., ``stack(unstack(x,
+    axis=axis), axis=axis) == x``.
+
+    .. versionadded:: 2.1.0
+
+    Parameters
+    ----------
+    x : ndarray
+        The array to be unstacked.
+
+    Returns
+    -------
+    unstacked : tuple of ndarrays
+        The unstacked arrays.
+
+    See Also
+    --------
+    stack : Join a sequence of arrays along a new axis.
+    concatenate : Join a sequence of arrays along an existing axis.
+    vstack : Stack arrays in sequence vertically (row wise).
+    hstack : Stack arrays in sequence horizontally (column wise).
+    dstack : Stack arrays in sequence depth wise (along third axis).
+    column_stack : Stack 1-D arrays as columns into a 2-D array.
+    vsplit : Split an array into multiple sub-arrays vertically (row-wise).
+    unstack : Split an array into a tuple of sub-arrays along an axis.
+    block : Assemble an nd-array from nested lists of blocks.
+    split : Split array into a list of multiple sub-arrays of equal size.
+
+    Examples
+    --------
+    >>> arr = np.arange(24).reshape((2, 3, 4))
+    >>> np.unstack(arr)
+    (array([[ 0,  1,  2,  3],
+            [ 4,  5,  6,  7],
+            [ 8,  9, 10, 11]]),
+     array([[12, 13, 14, 15],
+            [16, 17, 18, 19],
+            [20, 21, 22, 23]]))
+    >>> np.unstack(arr, axis=1)
+    (array([[ 0,  1,  2,  3],
+            [12, 13, 14, 15]]),
+     array([[ 4,  5,  6,  7],
+            [16, 17, 18, 19]]),
+     array([[ 8,  9, 10, 11],
+            [20, 21, 22, 23]]))
+    >>> arr2 = np.stack(np.unstack(arr, axis=1), axis=1)
+    >>> arr2.shape
+    (2, 3, 4)
+    >>> np.all(arr == arr2)
+    np.True_
+
+    """
+    x = asanyarray(x)
+
+    axis = normalize_axis_index(axis, x.ndim)
```

Comment:
In principle, one can just do `return tuple(np.rollaxis(x, axis))`.

Reply:
Right, this whole function is just `tuple(np.moveaxis(x, axis, 0))`.

There is no reason to have it NumPy beyond torch not supporting the `tuple()` I think.
(@rgommers let's make sure, because if they do it seems unnecessary.  This is a function that if we did customize `ndarray.__array_namespace__`, I may not include in NumPy.

Although it is also one of those functions that can be useful for discoverability (using moveaxis like that is obvious to us, but not for every user)!  In that sense, I do think it may be a nice teaching opportunity to say that it is equivalent!

Reply:
> There is no reason to have it NumPy beyond torch not supporting the `tuple()` I think.

https://github.com/data-apis/array-api/issues/487 covers reasons, as pointed out by @asmeurer.  

> I do think it may be a nice teaching opportunity to say that it is equivalent!

That might be useful in the Notes section for a few people, but I don't think we should be encouraging using `tuple()` on an array at all (nor `len()` for that matter).

Reply:
Ah, right, there is always the argument that array iteration isn't necessary well defined (i.e. my `iteraxis()` thoughts, also)!